### PR TITLE
bugfix in source-dependent analysis

### PR DESCRIPTION
**Pull request by SeiyaNozaki**
I found some small bugs in the recent pull request for source dependent analysis.
So this PR fixed them.

### CODE REVIEW

File: lstchain/scripts/lstchain_dl1_to_dl2.py [hunk 1]
```diff
@@ -67,7 +67,9 @@ def main():
     data = pd.read_hdf(args.datafile, key=dl1_params_lstcam_key)
 
     if config['source_dependent']:
-        data = pd.concat([data, pd.read_hdf(data, key=dl1_params_src_dep_lstcam_key)], axis=1)
+        data = pd.concat([data, pd.read_hdf(args.datafile, key=dl1_params_src_dep_lstcam_key)], axis=1)
+        data = data.drop("index", axis=1)
```
Comment (MaxNoe) [inline lstchain/scripts/lstchain_dl1_to_dl2.py:71]: Why do you need to drop "index" here?

I think if there is a column "index" in the file, that is not recognized as **the** index when reading it with pandas, either the thing writing it should make something different.

Comment (SeiyaNozaki) [reply lstchain/scripts/lstchain_dl1_to_dl2.py:71]: After adding the source parameters using `lstchain_add_source_dependent_parameters.py`, `index` column is apparently included in `dl1_params_src_dep_lstcam_key`.
Actually `write_dataframe` saved this `index` column.

dl1_src_dep

Maybe due to this `index` column, I got the following error  during ` write_dl2_dataframe` in `lstchain_dl1_to_dl2.py`
```
ValueError: appended items do not match existing items in table!
```

Do you have any suggestions...?

Comment (MaxNoe) [reply lstchain/scripts/lstchain_dl1_to_dl2.py:71]: Most pandas write methods have an `index` keyword argument, that defines if the index should be stored to file. 

Normally you want that, and normally pandas detects that again on reading the file and automatically assigns that column as the `index` of the `DataFrame` you get. 

Somehow, it seems to be duplicated here (observe how the column `index` is identical to the actual index of the DataFrame.

A question would be, if the DataFrame you pass to `write_dataframe` already has the `index` column or if the `write_dataframe` function calls `reset_index`.

Comment (MaxNoe) [reply lstchain/scripts/lstchain_dl1_to_dl2.py:71]: In any case, that column should not be in the file as a column.

Comment (SeiyaNozaki) [reply lstchain/scripts/lstchain_dl1_to_dl2.py]: I fixed it by using `set_index` when reading.

**Comment by rlopezcoto** [comment]
This one looks good now, is it ok @MaxNoe or are you planning to work on a better solution also related to #375? In any case I would accept this fix by now

**Comment by MaxNoe** [comment]
Let's merge, I'll make a PR adressing the hdf5 stuff also fixing the index stuff

**Comment by MaxNoe** [comment]
Actually, there is 0 % test coverage here.

Can we test dl1_to_dl2 with `source_dependent=True`?

**Comment by rlopezcoto** [comment]
> Can we test dl1_to_dl2 with source_dependent=True?

Do you want to include the test in this PR? We definitely need to improve the test coverage in general...

**Pull request merged**